framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4240,6 +4291,202 @@ private void makeGroundTargetType(
+                                != typeToCapture.getTypeArguments().size())) {
+            return false;
+        }
+
+        for (int i = 0; i < capturedTypeMirror.getTypeArguments().size(); i++) {
+            if (TypesUtils.isCaptured(capturedTypeMirror.getTypeArguments().get(i))
+                    && typeToCapture.getTypeArguments().get(i).getKind() == TypeKind.WILDCARD) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Apply capture conversion.
+     *
+     * <p>Capture conversion is the process of converting wildcards in a generic type to fresh type
+     * variables. See <a
+     * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-5.html#jls-5.1.10">JLS
+     * 5.1.10</a> for all the details.
+     *
+     * <p>If {@code type} is not a declared type or if it does not have any wildcard type arguments,
+     * this method returns {@code type}.
+     *
+     * @param typeToCapture type to capture
+     * @return the captured type
+     */
+    public AnnotatedTypeMirror applyCaptureConversion(AnnotatedTypeMirror typeToCapture) {
+        TypeMirror capturedTypeMirror = types.capture(typeToCapture.getUnderlyingType());
+        return applyCaptureConversion(typeToCapture, capturedTypeMirror);
+    }
+
+    /**
+     * Apply capture conversion.
+     *
+     * <p>Capture conversion is the process of converting wildcards in a generic type to fresh type
+     * variables. See <a
+     * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-5.html#jls-5.1.10">JLS
+     * 5.1.10</a> for all the details.
+     *
+     * <p>If {@code type} is not a declared type or if it does not have any wildcard type arguments,
+     * this method returns {@code type}.
+     *
+     * @param type type to capture
+     * @param typeMirror type from the Java compiler that has captured wildcards, which are used as
+     *     the underlying type of the captured wildcards
+     * @return the captured type
+     */
+    public AnnotatedTypeMirror applyCaptureConversion(
+            AnnotatedTypeMirror type, TypeMirror typeMirror) {
+        if (!shouldCapture(type, typeMirror)) {
+            return type;
+        }
+
+        DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+        AnnotatedDeclaredType typeToCapture = (AnnotatedDeclaredType) type;
+        AnnotatedDeclaredType capturedType =
+                (AnnotatedDeclaredType)
+                        AnnotatedTypeMirror.createType(capturedTypeMirror, this, false);
+        capturedType.getTypeArguments(); // initialize the type arguments
+
+        AnnotatedDeclaredType typeDeclaration =
+                (AnnotatedDeclaredType)
+                        getAnnotatedType(typeToCapture.getUnderlyingType().asElement());
+
+        // Create a mapping from type variables in typeDeclaration to type arguments.
+        Map<TypeVariable, AnnotatedTypeMirror> typeVarToTypeArguments = new HashMap<>();
```

Comment:
What is the distinction between these two variables, which have the same `Map` type?  Does the comment apply to just one of them or to both?